src/libostree/ostree-repo-private.h

```diff
@@ -234,9 +234,16 @@ typedef OstreeRepo _OstreeRepoAutoTransaction;
 static inline void
 _ostree_repo_auto_transaction_cleanup (void *p)
 {
+  if (p == NULL || !OSTREE_IS_REPO(p))
```

Comment:
I think these are two different things; if the `OSTREE_IS_REPO(p)` check fails then there must be e.g. memory corruption involved.

How about:

```
if (p == NULL)
  return;
g_return_if_fail (OSTREE_IS_REPO (p));
```

Now in theory, API entrypoints like `ostree_repo_abort_transaction()` could also be doing the dynamic type check so we could drop it here, but I haven't bothered with that in most of our code so far.

Reply:
Thanks for the feedback, it looks indeed better, I'll update that way.
For completeness, I added this check because if this function becomes public API there is an higher chance of users mistakenly calling this with a wrong parameter, so some basic sanity check seems worth.